dataprofiler/tests/profilers/test_profile_builder.py

```diff
@@ -3790,6 +3855,326 @@ def test_null_row_stats_correct_after_updates(self, *mocks):
+            }
+        )
+        profiler = StructuredProfiler(pd.DataFrame([]), options=profiler_options_1)
+        self.assertIsNone(profiler._get_duplicate_row_count())
+
+    def test_duplicate_row_count_empty_profiler(self):
+        profiler_options = ProfilerOptions()
+        profiler_options.set(
+            {
+                "*.is_enabled": False,
+                "row_statistics.is_enabled": True,
+            }
+        )
+        profiler = StructuredProfiler(pd.DataFrame([]), options=profiler_options)
+        self.assertEqual(0, profiler._get_duplicate_row_count())
+
+    def test_duplicate_row_count_hll_cardinality_greater_than_total_samples(self):
+        profiler_options = ProfilerOptions()
+        profiler_options.set(
+            {
+                "*.is_enabled": False,
+                "row_statistics.is_enabled": True,
+                "row_statistics.unique_count.is_enabled": True,
+                "row_statistics.unique_count.hashing_method": "hll",
+            }
+        )
+        with mock.patch(
+            "dataprofiler.profilers.profile_builder.HyperLogLog",
+            spec=dataprofiler.profilers.profile_builder.HyperLogLog,
+        ) as hll_mock:
+            hll_mock.return_value.cardinality.return_value = 1000
+            profiler = StructuredProfiler(pd.DataFrame([]), options=profiler_options)
+
+        self.assertEqual(1000, profiler.hashed_row_object.cardinality())
+        self.assertEqual(0, profiler._get_duplicate_row_count())
+
+    def test_save_and_load_hll(self):
+        self.assertEqual(15, self.trained_schema_hll.hashed_row_object.cardinality())
+
+        # Save and Load profile with Mock IO
+        with mock.patch("builtins.open") as m:
+            mock_file = setup_save_mock_open(m)
+            self.trained_schema_hll.save()
+
+            mock_file.seek(0)
+            with mock.patch("dataprofiler.profilers.profile_builder." "DataLabeler"):
+                load_profile = dp.StructuredProfiler.load("mock.pkl")
+        self.assertEqual(15, load_profile.hashed_row_object.cardinality())
+
+        # Check that reports are equivalent
+        save_report = test_utils.clean_report(self.trained_schema_hll.report())
+        load_report = test_utils.clean_report(load_profile.report())
+        self.assertEqual(
+            save_report["global_stats"]["unique_row_ratio"],
+            load_report["global_stats"]["unique_row_ratio"],
+        )
+
+        self.assertEqual(
+            save_report["global_stats"]["duplicate_row_count"],
+            load_report["global_stats"]["duplicate_row_count"],
+        )
+
+        # validate both are still usable after
+        self.trained_schema_hll.update_profile(
+            pd.DataFrame({"names": ["hello"], "numbers": [5], "tf_null": [None]})
+        )
+        load_profile.update_profile(
+            pd.DataFrame({"names": ["hello"], "numbers": [5], "tf_null": [None]})
+        )
```

Comment:
this is good, should be the same update to each. However, I suggest having one row be new and one be a duplicate such that we can validate it updates as expected and handles both conditions after serialization.